### PR TITLE
Fix filters on tests and boots tab

### PR DESCRIPTION
**Pull request by loadez**

### CODE REVIEW

File: backend/kernelCI_app/views/treeDetailsView.py [hunk 1]
```diff
@@ -85,12 +86,16 @@ def get(self, request, commit_hash):
         for f in filter_params.filters:
             field = f['field']
             table = None
+            if field == "buildStatus":
```
Comment (lfjnascimento) [inline backend/kernelCI_app/views/treeDetailsView.py:89]: this should be done in the frontend, in the fetch function

File: backend/kernelCI_app/views/treeTestsView.py [hunk 2]
```diff
@@ -38,15 +39,59 @@ def extract_error_message(self, misc: Union[str, dict, None]):
...
         return "unknown error"
 
-    def get(self, request, commit_hash: str | None):
-        path_param = request.GET.get('path')
+    def __mapStatusInClause(self, value):
+        if value is not list:
+            value = [value,]
+        status = []
+        for i in value:
+            if i == "Success":
```
Comment (lfjnascimento) [inline backend/kernelCI_app/views/treeTestsView.py:47]: test status should not be grouped, we must have a filter for each status (MISS, DONE, SKIP....)

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "cSpell.words": [
     "tanstack"
-  ]
+  ],
+  "python.analysis.typeCheckingMode": "basic"
```
Comment (lfjnascimento) [inline .vscode/settings.json:5]: please, remove .vscode and put it in .gitignore

File: backend/kernelCI_app/utils.py [hunk 4]
```diff
@@ -43,6 +43,8 @@ class FilterParams():
     def __init__(self, request):
         self.filters = []
         self.create_filters_from_req(request)
+        for f in self.filters:
+            print(f)
```
Comment (lfjnascimento) [inline backend/kernelCI_app/utils.py:47]: leftover

File: backend/kernelCI_app/views/treeDetailsView.py [hunk 5]
```diff
@@ -89,8 +90,13 @@ def get(self, request, commit_hash):
                 table = 'builds'
             elif field in checkout_fields:
                 table = 'checkouts'
+            elif field == 'buildStatus':
+                table = 'builds'
+                field = 'valid'
+                f['value'] = f['value'] == 'Success'
```
Comment (lfjnascimento) [inline backend/kernelCI_app/views/treeDetailsView.py:96]: ?

File: dashboard/src/pages/TreeDetails/TreeDetails.tsx [hunk 6]
```diff
@@ -101,7 +101,7 @@ function TreeDetails(): JSX.Element {
   });
 
   const reqFilter = mapFilterToReq(diffFilter);
-
+  console.log(reqFilter);
```
Comment (lfjnascimento) [inline dashboard/src/pages/TreeDetails/TreeDetails.tsx:104]: left over

File: dashboard/src/pages/TreeDetails/TreeDetailsFilter.tsx [hunk 7]
```diff
@@ -71,14 +74,20 @@ export const mapFilterToReq = (
       filterMapped[reqField] = [values.toString()];
     }
   });
-
+  console.log(filterMapped);
```
Comment (lfjnascimento) [inline dashboard/src/pages/TreeDetails/TreeDetailsFilter.tsx:77]: left over

File: backend/kernelCI_app/views/treeDetailsView.py (hunk 5 above)
Comment (loadez) [reply backend/kernelCI_app/views/treeDetailsView.py:96]: This maps the value that we have on the front-end to the one that we use on the database

File: dashboard/src/pages/TreeDetails/Tabs/Boots/BootsTab.tsx [hunk 8]
```diff
@@ -16,13 +16,18 @@ import {
...
+
+const BootsTab = ({ reqFilter }: BootsTabProps): JSX.Element => {
   const { treeId } = useParams({
     from: '/tree/$treeId/',
   });
-  const { isLoading, data, error } = useBootsTab(treeId ?? '');
+  const { isLoading, data, error } = useBootsTab(treeId ?? '', reqFilter);
 
   if (error || !treeId) {
+    console.log('error:', error);
```
Comment (mari1912) [inline dashboard/src/pages/TreeDetails/Tabs/Boots/BootsTab.tsx:30]: nit: remove it

**Review by lfjnascimento — APPROVED** [review]

**Review by mari1912 — APPROVED** [review]

**Pull request merged**